clouddrift/pairs.py

```diff
@@ -0,0 +1,508 @@
+                )
+                for j, k in pairs
+            ]
+        else:
+            futures = [
+                executor.submit(
+                    chance_pair,
+                    lon[i[j] : i[j + 1]],
+                    lat[i[j] : i[j + 1]],
+                    lon[i[k] : i[k + 1]],
+                    lat[i[k] : i[k + 1]],
+                    time[i[j] : i[j + 1]],
+                    time[i[k] : i[k + 1]],
+                    space_tolerance,
+                    time_tolerance,
+                )
+                for j, k in pairs
+            ]
+        for future in as_completed(futures):
+            res = future.result()
+            # chance_pair function returns empty arrays if no chance criteria
+            # are satisfied. We only want to keep pairs that satisfy the
+            # criteria. chance_pair returns a tuple of arrays that are always
+            # the same size, so we only need to check the length of the first
+            # array.
+            if res[0].size > 0:
+                results.append((pairs[futures.index(future)], res))
+    return results
+
+
+def pair_bounding_box_overlap(
+    lon1: array_like,
+    lat1: array_like,
+    lon2: array_like,
+    lat2: array_like,
+    tolerance: Optional[float] = 0,
+) -> Tuple[np.ndarray[bool], np.ndarray[bool]]:
+    """Given two arrays of longitudes and latitudes, return boolean masks for
+    their overlapping bounding boxes.
+
+    Parameters
+    ----------
+    lon1 : array_like
+        First array of longitudes in degrees.
+    lat1 : array_like
+        First array of latitudes in degrees.
+    lon2 : array_like
+        Second array of longitudes in degrees.
+    lat2 : array_like
+        Second array of latitudes in degrees.
+    tolerance : float, optional
+        Tolerance in degrees for the overlap. If the overlap is within this
+        tolerance, the bounding boxes are considered to overlap. Default is 0,
+        or no tolerance.
+
+    Returns
+    -------
+    overlap1 : np.ndarray[bool]
+        Boolean mask for ``lon1`` and ``lat1`` that indicates where their bounding
+        box overlaps with that of ``lon2`` and ``lat2``.
+    overlap2 : np.ndarray[bool]
+        Boolean mask for ``lon2`` and ``lat2`` that indicates where their bounding
+        box overlaps with that of ``lon1`` and ``lat1``.
+
+    Examples
+    --------
+    >>> lon1 = [0, 0, 1, 1]
+    >>> lat1 = [0, 0, 1, 1]
+    >>> lon2 = [1, 1, 2, 2]
+    >>> lat2 = [1, 1, 2, 2]
+    >>> pair_bounding_box_overlap(lon1, lat1, lon2, lat2, 0.5)
+    (array([False, False,  True,  True]), array([ True,  True, False, False]))
+    """
+    # First get the bounding box of each trajectory.
+    lon1_min, lon1_max = np.min(lon1), np.max(lon1)
+    lat1_min, lat1_max = np.min(lat1), np.max(lat1)
+    lon2_min, lon2_max = np.min(lon2), np.max(lon2)
+    lat2_min, lat2_max = np.min(lat2), np.max(lat2)
+
+    # TODO handle trajectories that cross the dateline
```

Comment:
I think you could do: 

```
lon1 = np.unwrap(lon1, period=360)
lon2 = np.unwrap(lon2, period=360)
```

It does the following:
```
lon = np.array([175, 179, -179, -170])
np.unwrap(lon, period=360)
>> array([175, 179, 181, 190])
```

Then at the end we could use:
```
def wrap(s, period):
     return np.mod(s + period/2, period) - period/2
```
to bring it back to [-180, 180] or modify it slightly to bring it back to [0, 360].
```